src/sentry/south_migrations/0513_fix_gh_bb_repo_data.py

```diff
@@ -0,0 +1,1700 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+from django.db.models import F, Field, Func, Lookup, Q, Value
+
+# This is not strictly necessary, since we're not dealing with nulls in any of
+# the three spots we're interested in - name is required, and (at least for
+# GitHub/BitBucket) only two totally broken repos have null urls and/or config
+# names, and those repos take a different kind of repair which we're not doing
+# here. Realized this after the fact, and it doesn't break anything, so leaving
+# it here for posterity and future folks (like me) who are learning by example.
+class NotDistinctFrom(Lookup):
+    lookup_name = 'notdistinctfrom'
+
+    def as_sql(self, compiler, connection):
+        lhs, lhs_params = self.process_lhs(compiler, connection)
+        rhs, rhs_params = self.process_rhs(compiler, connection)
+        params = lhs_params + rhs_params
+        return ('%s IS NOT DISTINCT FROM %s' % (lhs, rhs), params)
+
+Field.register_lookup(NotDistinctFrom)
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        """
+        Before https://github.com/getsentry/sentry/pull/12290, for GitHub and
+        BitBucket repos we were only updating `repo.config['name']` when repo data
+        in webhook events was different than what we had, while *not* updating
+        `repo.name` or `repo.url`.
+
+        Since that was merged, webhooks have fixed some data but not all. This
+        updates all GitHub or BitBucket repos so that all data matches `config['name]`.
+
+        See comment below for testing.
+        """
+
+        # functions, etc for annotating, filtering, and fixing Repository records
```

Comment:
Could remove this since there aren't any.

Reply:
Aren't any what? (Sorry, maybe I'm being slow.)

Reply:
I got tripped up on 'functions' I was thought it meant python functions, but I'm a goof.

Reply:
Okay, updated the comment to hopefully make it a little clearer